WordPress/src/main/java/org/wordpress/android/util/DialogUtils.java

```diff
@@ -11,22 +14,55 @@
 import org.wordpress.android.widgets.OpenSansEditText;
 import org.wordpress.android.widgets.WPTextView;
 
-public class DialogUtils {
-    public static void showMyProfileDialog(Context context,
-                                           String title,
-                                           String initialText,
-                                           String hint,
-                                           boolean isMultiline,
-                                           final Callback callback) {
-        LayoutInflater layoutInflater = LayoutInflater.from(context);
+public class DialogUtils extends DialogFragment {
+
+    private static final String TITLE_TAG = "TITLE";
+    private static final String INITIAL_TEXT_TAG = "INITIAL-TEXT";
+    private static final String HINT_TAG = "HINT";
+    private static final String IS_MULTILINE_TAG = "IS-MULTILINE";
+
+    private static Context mContext;
+    private static Callback mCallback;
+
+    public static DialogUtils newInstance(Context context,
+                                   String title,
+                                   String initialText,
+                                   String hint,
+                                   boolean isMultiline,
+                                   Callback callback) {
+
+        DialogUtils dialogUtils = new DialogUtils();
+        Bundle args = new Bundle();
+
+        mContext = context;
+        mCallback = callback;
+
+        args.putString(TITLE_TAG, title);
+        args.putString(INITIAL_TEXT_TAG, initialText);
+        args.putString(HINT_TAG, hint);
+        args.putBoolean(IS_MULTILINE_TAG, isMultiline);
+
+        dialogUtils.setArguments(args);
+        return dialogUtils;
+    }
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
```

Comment:
AFAIK, there is no particular need to pass the Context via a state variable. You can just use `getActivity()` to get a hold to the needed Context. Cheers!